test/ruby/test_method.rb

```diff
@@ -1181,6 +1181,27 @@ def test_argument_error_location
     assert_separately [], "RubyVM::InstructionSequence.compile_option = {trace_instruction: false}\n" + body
   end
 
+  def test_zsuper_private_override_instance_method
+    assert_separately(%w(--disable-gems), <<-'end;', timeout: 30)
+      # Bug #16942 [ruby-core:98691]
+      module M
+        def x
+        end
+      end
+
+      module M2
+        prepend Module.new
+        include M
+        private :x
+      end
+
+      ::Object.prepend(M2)
+
+       m = Object.instance_method(:x)
```

Comment:
Indents slipping?

Reply:
Yep, sorry about that, pushed a commit to fix that.